document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -263,8 +269,166 @@ public BulkUpdateResult bulkUpdateSubDocs(Map<Key, Map<String, Document>> docume
   }
 
   @Override
-  public BulkUpdateResult bulkOperationOnArrayValue(BulkArrayValueUpdateRequest request) {
-    throw new UnsupportedOperationException();
+  public BulkUpdateResult bulkOperationOnArrayValue(BulkArrayValueUpdateRequest request)
+      throws Exception {
+    Operation operation = request.getOperation();
+    switch (operation) {
+        // todo: All of these implementations do a GET -> in memory update -> UPSERT which is not
+        // optimal. Changes this logic to use a single query using jsob_set and json_insert
+      case ADD:
+        return addImpl(request);
```

Comment:
nit:For readability, we can rename the function names>

`addImpl` -> `bulkAddOnArrayValue`
`setImpl` -> `bulkSetOnArrayValue`
`removeImpl` -> bulkRemoveOnArrayValue`

above suggestion is based on the pattern - bulk<Operation>OnArrayValue

What do you think?